include/igl/min_quad_with_fixed.cpp

```diff
@@ -582,9 +583,330 @@ IGL_INLINE bool igl::min_quad_with_fixed(
+    Eigen::Matrix<Scalar,nn,1> ff(dyn_nn);
+    ff.head(dyn_n) =  f;
+    ff.tail(dyn_m) = -b;
+    return ff;
+  };
+  const Eigen::Matrix<Scalar,nn,1> ff = make_ff();
+  const auto make_kk  = [&]()
+  {
+    // Windows can't remember that nn is const.
+    constexpr const int nn = n == Eigen::Dynamic ? Eigen::Dynamic : n+m;
+    Eigen::Array<bool,nn,1> kk =
+      Eigen::Array<bool,nn,1>::Constant(dyn_nn,1,false);
+    kk.head(dyn_n) =  k;
+    return kk;
+  };
+  const Eigen::Array<bool,nn,1> kk = make_kk();
+  const auto make_bcbc= [&]()
+  {
+    // Windows can't remember that nn is const.
+    constexpr const int nn = n == Eigen::Dynamic ? Eigen::Dynamic : n+m;
+    Eigen::Matrix<Scalar,nn,1> bcbc(dyn_nn);
+    bcbc.head(dyn_n) =  bc;
+    return bcbc;
+  };
+  const Eigen::Matrix<double,nn,1> bcbc = make_bcbc();
+  const Eigen::Matrix<Scalar,nn,1> xx = 
+    min_quad_with_fixed<Scalar,nn,false>(HH,ff,kk,bcbc);
+  //std::cout<<igl::matlab_format(xx.head(dyn_n).eval(),"x")<<std::endl;
+  return xx.head(dyn_n);
+}
+
+template <typename Scalar, int n, bool Hpd>
+IGL_INLINE Eigen::Matrix<Scalar,n,1> igl::min_quad_with_fixed(
+  const Eigen::Matrix<Scalar,n,n> & H,
+  const Eigen::Matrix<Scalar,n,1> & f,
+  const Eigen::Array<bool,n,1> & k,
+  const Eigen::Matrix<Scalar,n,1> & bc)
+{
+  //printf("igl::min_quad_with_fixed<S,n,Hpd>\n");
+  //std::cout<<igl::matlab_format(H,"H")<<std::endl;
+  //std::cout<<igl::matlab_format(f,"f")<<std::endl;
+  //std::cout<<igl::matlab_format(k,"k")<<std::endl;
+  //std::cout<<igl::matlab_format(bc,"bc")<<std::endl;
+  assert(H.isApprox(H.transpose(),1e-7)); 
+  assert(H.rows() == H.cols());
+  assert(H.rows() == f.size());
+  assert(H.rows() == k.size());
+  assert(H.rows() == bc.size());
+  const auto kcount = k.count();
+  // Everything fixed
+  if(kcount == (Eigen::Dynamic?H.rows():n))
+  {
+    //std::cout<<igl::matlab_format(bc,"x")<<std::endl;
+    return bc;
+  }
+  // Nothing fixed
+  if(kcount == 0)
+  {
+    // avoid function call
+    typedef Eigen::Matrix<Scalar,n,n> MatrixSn;
+    typedef typename 
+      std::conditional<Hpd,Eigen::LLT<MatrixSn>,Eigen::CompleteOrthogonalDecomposition<MatrixSn>>::type
+      Solver;
+    //std::cout<<igl::matlab_format( Solver(H).solve(-f).eval() ,"x")<<std::endl;
+    return Solver(H).solve(-f);
+  }
+  // All-but-one fixed
+  if( (Eigen::Dynamic?H.rows():n)-kcount == 1)
+  {
+    // which one is not fixed?
+    int u = -1;
+    for(int i=0;i<k.size();i++){ if(!k(i)){ u=i; break; } } 
+    assert(u>=0);
+    // min ½ x(u) Huu x(u) + x(u)(fu + H(u,k)bc(k))
+    // Huu x(u) = -(fu + H(u,k) bc(k))
+    // x(u) = (-fu + ∑ -Huj bcj)/Huu
+    Eigen::Matrix<Scalar,n,1> x = bc;
+    x(u) = -f(u);
+    for(int i=0;i<k.size();i++){ if(i!=u){ x(u)-=bc(i)*H(i,u); } } 
+    x(u) /= H(u,u);
+    //std::cout<<igl::matlab_format(x,"x")<<std::endl;
+    return x;
+  }
+  // Is there a smart template way to do this?
```

Comment:
I guess you could do a templated for-loop starting from 16, and dispatching to the appropriate templated function when the argument matches (with a fallback to the dynamic version). Cf this example:
https://gist.github.com/disconnect3d/13c2d035bb31b244df14

Reply:
whoa. This is cool. The nightmare with windows and the templated code make not want to go near this for a while. I will add your note as a comment for the future.